src/data/set/intervals/monotone.lean

```diff
@@ -177,3 +178,46 @@ calc (⋃ x, Ioo (f x) (g x)) = (⋃ x, Ioi (f x)) ∩ ⋃ x, Iio (g x) :
 ... = Ioi a ∩ Iio b : congr_arg2 (∩) ha.Union_Ioi_eq hb.Union_Iio_eq
 
 end Union
+
+section succ_order
+
+open order
+
+variables {α : Type*} [partial_order α] [order_bot α]
+  [succ_order α] [is_succ_archimedean α] [no_max_order α] {n : α} {φ : α → α}
+
+lemma strict_mono_on.Iic_id_le (hφ : strict_mono_on φ (set.Iic n)) :
+  ∀ m ≤ n, m ≤ φ m :=
+begin
+  revert hφ,
+  refine succ.rec_bot (λ n, strict_mono_on φ (set.Iic n) → ∀ m, m ≤ n → m ≤ φ m) _ _ _,
+  { simp_rw le_bot_iff,
+    rintro _ m rfl,
+    exact bot_le },
+  { rintro k ih hφ m hm,
+    obtain (rfl | h) := le_succ_iff_eq_or_le.1 hm,
+    { specialize ih (strict_mono_on.mono hφ (λ x hx, le_trans hx (le_succ _))) k le_rfl,
+      exact le_trans (succ_mono ih) (succ_le_of_lt (hφ (le_succ _) le_rfl (lt_succ _))) },
+    { exact ih (strict_mono_on.mono hφ (λ x hx, le_trans hx (le_succ _))) _ h } }
+end
+
+lemma strict_mono_on_Iic_of_lt_succ (hφ : ∀ m, succ m ≤ n → φ m < φ (succ m)) :
+  strict_mono_on φ (set.Iic n) :=
```

Comment:
This is true for any codomain, not only for self-maps. Again, you don't need `no_max_order` because `φ m < φ (succ m)` implies `succ m ≠ m`.